astropy/wcs/setup_package.py

```diff
@@ -286,13 +283,13 @@ def get_extensions():
     ]
 
     wcslib_config_paths = [
-        join(WCSROOT, "include", "astropy_wcs", "wcsconfig.h"),
-        join(WCSROOT, "include", "wcsconfig.h"),
+        WCSROOT / "include" / "astropy_wcs" / "wcsconfig.h",
+        WCSROOT / "include" / "wcsconfig.h",
     ]
 
     get_wcslib_cfg(cfg, wcslib_files, wcslib_config_paths)
 
-    cfg["include_dirs"].append(join(WCSROOT, "include"))
+    cfg["include_dirs"].append(WCSROOT / "include")
```

Comment:
Should this have a `str(...)` too, like the others?

Reply:
ah, yes. I got things mixed up here, but passing `Path`s directly to setuptools requires another upgrade that we cannot afford yet because it's not available on exotic archs.
Anyway, thanks for catching this !

Reply:
Hmm, that leaves me still confused what part actually is solved by upgrading extension helpers.

Reply:
it should be clearer if you look just at the second commit
[`8b4fc9f` (#17193)](https://github.com/astropy/astropy/pull/17193/commits/8b4fc9fcd897b0e2599ec388ad87e49efe075161)